CNV-80383: Add 5 CNV release notes for 4.22 (#112604)
openshift/openshift-docs: virt/release_notes/virt-4-22-release-notes.adoc

// Infrastructure

Monitor node memory overcommit level for virtual machines::
+

> Cluster administrators can balance workloads using the VM memory overcommit and utilization dashboard. The dashboard allows you to monitor whether the clusters are underutilized or at risk due to memory overcommit to decide whether to expand the cluster.

Reviewer: ```suggestion
Cluster administrators can balance workloads using the VM memory overcommit and utilization dashboard. The dashboard allows you to monitor whether clusters are underutilized or at risk due to memory overcommit, or to decide whether to expand a cluster.
```

// Storage

Native storage migration support for virtual machines::
+

> {VirtProductName} supports native storage migration in {product-title} 4.20 and later. Before this update, you installed Migration Toolkit for Containers (MTC) to migrate virtual machine (VM) disks to different storage classes. You need not install MTC since the VM disk storage class migration is native to {VirtProductName}. Additionally, it is no longer a requirement that the VMs you select for each bulk migration be in the same namespace.

Reviewer: ```suggestion
{VirtProductName} supports native storage migration in {product-title} 4.20 and later. Before this update, you installed Migration Toolkit for Containers (MTC) to migrate virtual machine (VM) disks to different storage classes. You no longer need to install MTC, since the VM disk storage class migration is native to {VirtProductName}. Additionally, it is no longer a requirement that the VMs you select for each bulk migration be in the same namespace.
```

+
link:https://redhat.atlassian.net/browse/CNV-81577[CNV-81577]

//CNV-73392
Create virtual machines from in-cluster native templates (Technology Preview)::
+

> Virtual machine (VM) owners can create VMs from {VirtProductName} cluster native template custom resource. The VM template tracks a golden image that is updated periodically, while reducing errors, and ensuring uniformity in the virtualized environment. You can host the template in all namespaces that you can control. This capability is a Technology Preview feature.

Reviewer: ```suggestion
Virtual machine (VM) owners can create VMs from the {VirtProductName} cluster native template custom resource. The VM template tracks a golden image that is updated periodically, while reducing errors, and ensuring uniformity in the virtualized environment. You can host the template in all namespaces that you can control. This capability is a Technology Preview feature.
```

Reviewer: This is a bit confusing - what does "cluster native template custom resource" mean, is this referring to a specific CR, and if so can we use the name of the CR instead to make this clearer? Maybe I'm just not understanding the feature though 🙂

Author: no, it's unclear to me too honestly. i've made a [suggestion](https://github.com/openshift/openshift-docs/pull/112604#discussion_r3344248030) and tagged QE to re-review it. thanks!

> Virtual machine (VM) owners can create VMs from the {VirtProductName} cluster native template custom resource. The VM template tracks a golden image that is updated periodically, while reducing errors, and ensuring uniformity in the virtualized environment. You can host the template in all namespaces that you can control. This capability is a Technology Preview feature.

Author: ```suggestion
Virtual machine (VM) owners can create VMs from native `VirtualMachineTemplate` resources in the cluster. The VM template tracks a golden image that is updated periodically, while reducing errors, and ensuring uniformity in the virtualized environment. You can host the template in all namespaces that you can control. This capability is a Technology Preview feature.
```

Author: @0xFelix here is the small rewrite i'd like for you to review

Reviewer 2: @danielclowers i see lot of comma do we really need the, like 
The VM template tracks a golden image that is updated periodically, reducing errors and ensuring uniformity in the virtualized environment.

Reviewer 2: Technology Preview feature. is used twice L224 and 226

Reviewer 3: Apart from what Geetika said, looks good to me

Author: @geetikakay @0xFelix thanks for confirming it looks technically accurate. and i agree tech preview is mentioned too many times. i'll clean it up a bit. appreciate it!

[role="_abstract"]
Some linked Jira tickets are accessible only with Red Hat credentials.

//CNV-78892
Non-versioned HyperConverged commands default to v1 API::

> In this release, the v1 API for the HyperConverged custom resource (CR) is introduced as a served version in preparation for a future migration from v1beta1 to v1. Due to the way Kubernetes selects default API versions, non-versioned commands such as `oc get hco`, `oc edit hyperconverged`, and `oc patch hyperconverged` now default to the v1 API. As a consequence, these commands may behave unexpectedly or fail because the v1 API is not yet ready for production use.

Reviewer: ```suggestion
In this release, the v1 API for the `HyperConverged` custom resource (CR) is introduced, in preparation for a future migration from v1beta1 to v1. Due to the way Kubernetes selects default API versions, non-versioned commands such as `oc get hco`, `oc edit hyperconverged`, and `oc patch hyperconverged` now default to the v1 API. As a consequence, these commands may behave unexpectedly or fail because the v1 API is not yet ready for production use.
```

+

> To work around this problem, use the fully versioned type name `hyperconvergeds.v1beta1.hco.kubevirt.io` when running commands against the HyperConverged CR. For example, use `oc get hyperconvergeds.v1beta1.hco.kubevirt.io` instead of `oc get hco`. For the `oc explain` command, use the `--api-version` flag: `oc explain --api-version=hco.kubevirt.io/v1beta1 hco.spec`. As a result, commands target the v1beta1 API as intended.

Reviewer: ```suggestion
To work around this problem, use the fully versioned type name `hyperconvergeds.v1beta1.hco.kubevirt.io` when running commands against the `HyperConverged` CR. For example, use `oc get hyperconvergeds.v1beta1.hco.kubevirt.io` instead of `oc get hco`. For the `oc explain` command, use the `--api-version` flag: `oc explain --api-version=hco.kubevirt.io/v1beta1 hco.spec`. As a result, commands target the v1beta1 API as intended.
```
